merge.py

```diff
@@ -1,98 +1,320 @@
+            raise AttributeError('Expected work_queue to be of type '
+                                 'Queue.Queue.')
+        self.work_queue = work_queue
+
+    @abc.abstractmethod
+    def merge(self):
+        """merge spins forever, merging things off of the work queue.
+        """
+        return
+
+
+class GitMerger(Merger):
+    """GitMerger merges Git pull requests.
+    """
+
+    def merge(self):
+        """merge spins forever, merging things off of the work queue.
+
+        Raises:
+            AttributeError: if work items are not of type github_helper.GithubPR
+        """
+        sys.stdout = open(os.path.join('log', self.config['name'] +
+                                       '_merger_log.txt'), 'w')
+        while True:
+            pr = self.work_queue.get()
+            if not isinstance(pr, github_helper.GithubPR.__class__):
+                raise AttributeError('Expected items in work_queue to be of '
+                                     'type github_helper.GithubPR')
+            _print_flush('Starting merge process for #{}.'.format(pr.get_num()))
+            if self.merge_git_pr(pr):
+                _print_flush('Merge concluded satisfactorily. Moving on.')
+            else:
+                _print_flush('Merge did not conclude satisfactorily. Adding '
+                             'PR back to queue to be tried again.')
+                self.work_queue.put(pr)
+
+    def merge_git_pr(self, pr):
+        """merge_git_pr merges git pull requests.
+
+        Args:
+            pr: The GithubPR to merge.
+        Returns:
+            True if merge was concluded satisfactorily (merged successfully,
+            or failed due to supposed fault of the PR itself).
+            False if failure was due to an environmental issue and should be
+            retried.
+        """
+        repo = self.config['repository']
+        tmp_dir = TMP_DIR_FMT.format(dir='{}-{}'.format(repo, pr.get_num()))
+        try:
+            _set_up(tmp_dir)
+        except AssertionError:
+            pr.post_error('Error setting up - trying again.')
+            return False
+        try:
+            self.clone_configure(tmp_dir, pr)
+            pr_name = 'finish-pr-{}'.format(pr.get_num())
+            self.checkout(pr_name, tmp_dir, pr)
+            branch = '{}/{}'.format('apache', self.config['merge_branch'])
+            self.rebase(branch, tmp_dir, pr)
+            self.target_checkout(branch, tmp_dir, pr)
+            self.do_local_merge(pr_name, tmp_dir, pr)
+            self.verify(tmp_dir, pr)
+            self.push(tmp_dir, pr)
+            _clean_up(tmp_dir)
+            _print_flush('Merged successfully!')
+        except AssertionError as err:
+            _print_flush(err)
+        except EnvironmentError as err:
+            _print_flush(err)
+            return False
+        return True
+
+    def clone_configure(self, tmp_dir, pr):
+        """Clone repository and configure.
```

Comment:
configure what?

I'd propose fleshing this out to describe the steps taken and/or adding comments in line. Specifically:

Clone repository, configure the remotes, and fetch all upstream PRs